xdsl/dialects/riscv.py

```diff
@@ -155,6 +152,44 @@ def parse_parameter(parser: Parser) -> Register:
         name = parser.parse_optional_identifier()
         if name is None:
             return Register()
+        if not name.startswith("j"):
+            assert name in Register.RV32I_INDEX_BY_NAME.keys()
+        return Register(name)
+
+    def print_parameter(self, printer: Printer) -> None:
+        name = self.data.name
+        if name is None:
+            return
+        printer.print_string(name)
+
+    def verify(self) -> None:
+        if self.data.name is None or self.data.name.startswith("j"):
+            return
+        assert self.data.name in Register.RV32I_INDEX_BY_NAME.keys()
+
+
+@irdl_attr_definition
+class FloatingRegisterType(Data[Register], TypeAttribute):
```

Comment:
sorry for being pedantic, but can you use `Float` instead of `Floating`?
I want to remember one mnemonic if possible :)